src/trackers/broadcasthenet.py

```
@@ -259,26 +609,98 @@ async def upload(self, meta: Meta) -> bool:
+                            torrent_id = detail_match.group(2)
+                            break
+
+                    if not torrent_id:
+                        dl_match = re.search(r"torrents\.php\?action=download(?:&amp;|&)id=(\d+)", detail_response.text)
+                        if dl_match:
+                            torrent_id = dl_match.group(1)
+
+                    if not torrent_id:
+                        if not self.api_key:
+                            raise UploadError(
+                                f"BTN upload reached intermediate page but failed to resolve torrent_id via HTML. Set an api_key in your BTN config to enable API fallback.",
```

Ruff (0.16.1)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)